pkg/rootless/rootless_linux.go

```diff
@@ -294,6 +294,11 @@ func becomeRootInUserNS(pausePid, fileToRead string, fileOutput *os.File) (_ boo
 	}
 
 	if fileOutput != nil {
+		ret := C.reexec_in_user_namespace_wait(pidC, 0)
+		if ret < 0 {
+			return false, -1, errors.New("error waiting for the re-exec process")
+		}
+
```

Comment:
IDK, but will ask.  We have a couple of other returns before the end of this function at lines 311 and 315.  Do we need to call there too?

Reply:
I think we probably should.  Can we `defer` a function that runs `reexec_in_user_namespace_wait`?  Would there be a problem if we ran it against the same PID in rapid succession?

Reply:
@jdieter if we did the defer, we'd need to capture the status of the call and return appropriately based on that.  I'm not sure that's doable with defer, @mheon or @vrothberg might have some smarter thoughts on that or my question in general.

Reply:
we could probably record the exit error and run it only when retErr != nil.  What do you think?

Reply:
Makes sense to me.

Reply:
Ok, I'm not very proficient in Go, but tell me if my lastest push has what you're looking for.  I've setup a deferred function that waits for the child if `retErr != nil`.  I don't bother with the return value from `reexec_in_user_namespace_wait` because we presumably are more interested in passing whatever error is causing us to bail.